drivers/flash/nrf_qspi_nor.c

```diff
@@ -27,15 +28,13 @@ LOG_MODULE_REGISTER(qspi_nor, CONFIG_FLASH_LOG_LEVEL);
 #include <hal/nrf_gpio.h>
 
 struct qspi_nor_data {
-#if !defined(CONFIG_PM_DEVICE_RUNTIME) && defined(CONFIG_MULTITHREADING)
-	/* A semaphore to control QSPI deactivation. */
-	struct k_sem count;
-#endif
 #ifdef CONFIG_MULTITHREADING
 	/* The semaphore to control exclusive access to the device. */
 	struct k_sem sem;
 	/* The semaphore to indicate that transfer has completed. */
 	struct k_sem sync;
+	/* A counter to control QSPI deactivation. */
+	atomic_t usage_count;
```

Comment:
Why remove dependency on `!defined(CONFIG_PM_DEVICE_RUNTIME) && defined(CONFIG_MULTITHREADING)`?
I see this counter only exists to cover this subcase.

Reply:
The `CONFIG_MULTITHREADING` dependency is still there.
`CONFIG_PM_DEVICE_RUNTIME` is removed because the existence of that option is a global setting, not a local one.
Device runtime PM can be enabled or disabled arbitrarily at runtime.